### ISSUE TITLE
Make server not crash when there's a syntax error in server.js

### ISSUE THREAD

**Issue by michael** [substance/journal#44]
See tahi-journal.

Seems this is not enough error handling:

``` js
    browserify({ debug: true, cache: false })
      .add(path.join(__dirname, "client", "app.js"))
      .bundle()
      .pipe(res)
      .on('error', function(req, res, next) {
        res.status(500).send("Internal error.");
      });
```

I tried adding a try-catch but still i get:

```
http://127.0.0.1:5000/

events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: Parsing file /Users/michael/projects/plos/tahi-journal/client/writer/components/content_toolbar.js: Unexpected token (86:4)
    at Deps.parseDeps (/Users/michael/projects/plos/tahi-journal/node_modules/browserify/node_modu
```

**Comment by Integral** [comment]
@michael, did you mean syntax error inside browserify bundle?

**Comment by michael** [comment]
Exactly. Server crashes then.

**Comment by Integral** [comment]
OK, but how do you then know that it fails? Via terminal message?

**Comment by michael** [comment]
Ideally it outputs the error on the screen too, not only console.

**Comment by Integral** [comment]
@michael, why did you drop out browserify middleware? It is not crushes the server and you can use it like this:
`app.get('/app.js', browserify(__dirname + '/client/app.js'));`

**Comment by michael** [comment]
We wanted less overhead, because we don't use any smart caching and fole watching, that browserify-middleware implements. Oliver just made the change. It's not so tragic when the server crashes for now, but if you have an idea how to fix it, please try and push it if its working.

**Comment by Integral** [comment]
ok, you can do it like this https://github.com/substance/tahi-journal/commit/ce3da633127ceaa8cbf6e4ddc008dfc16cfa1de5
also you can get rid of try/catch and use just res.end()
it will print error in terminal console and in screen if you are trying to view js file, but not for template (because browserify works for js files), but perhaps you can write some script to send it to browser console

**Comment by Integral** [comment]
and of course it will not crash your server ;)

**Comment by michael** [comment]
bravo daniel! :)

seems that order of the handler did matter. Do we really need the try catch block inside the error handler?

**Comment by Integral** [comment]
yes, it did! because you need to attach error listner to bundle function ;)
we need to do try/catch either res.end() to close connection, otherwise express will crash server because of with few errors you will try to use res few times while you are already send response.
also when you are sending error you will somehow see it in browser (red message in console), but with res.end i'm not sure...

**Comment by michael** [comment]
I did this now. So it prints the error in the console in the browser also.

**Comment by michael** [comment]
https://github.com/substance/tahi-journal/commit/82d83596f839b83086177cf826b627ab1ac7c811

**Comment by Integral** [comment]
and now try http://localhost:5000/1
you will get express error
